addons/im_livechat/models/mail_channel.py

```diff
@@ -169,6 +172,34 @@ def _close_livechat_session(self, **kwargs):
             self.message_post(author_id=self.env.ref('base.partner_root').id,
                               body=self._get_visitor_leave_message(**kwargs), message_type='comment', subtype_xmlid='mail.mt_comment')
 
+    def _process_chatbot_next_step(self, step_id):
+        self.ensure_one()
+        if step_id.sequence <= self.livechat_chatbot_current_step_id.sequence:
+            raise ValidationError(_('The new step sequence must be higher than the old one.'))
+
+        # We change the current step to the new step
+        self.livechat_chatbot_current_step_id = step_id
+
+        self.with_context(mail_create_nosubscribe=True).message_post(
+            author_id=self.env.ref('base.partner_root').id,
+            body=step_id.message,
+            message_type='comment',
+            subtype_xmlid='mail.mt_comment',
+        )
+
+    def _message_post_after_hook(self, message, msg_vals):
```

Comment:
Since "message_post" returns the created message, do we really need this?

We could create this within "_process_chatbot_next_step", or maybe I missed something?

Let's try to keep things as simple as we can.